src/ui/view/settings/boxes.rs

```diff
@@ -0,0 +1,403 @@
+use iced::{Align, Column, Container, Length, Row};
+
+use crate::ui::{
+    component::{card, separation, text},
+    message::Message,
+};
+
+use crate::revaultd::config::{
+    Config, CosignerConfig, ManagerConfig, StakeholderConfig, WatchtowerConfig,
+};
+use bitcoin::{util::bip32, PublicKey};
+use std::{net::SocketAddr, path::PathBuf, vec::Vec};
+
+pub trait SettingsBox {
+    fn title(&self) -> &'static str;
+    fn description(&self) -> &'static str;
+    fn body<'a>(&self) -> Column<'a, Message>;
+    fn display<'a>(&self) -> Container<'a, Message> {
+        card::simple(Container::new(
+            Column::new()
+                .push(
+                    Row::new()
+                        .push(
+                            Container::new(
+                                Row::new()
+                                    .push(
+                                        Column::new()
+                                            .push(text::bold(text::simple(self.title())))
+                                            .push(text::small(self.description())),
+                                    )
+                                    .spacing(20),
+                            )
+                            .width(Length::Fill),
+                        )
+                        .spacing(20)
+                        .align_items(Align::Center),
+                )
+                .push(separation().width(Length::Fill))
+                .push(self.body())
+                .spacing(20),
+        ))
+        .width(Length::Fill)
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct SettingsBoxes {
+    pub general: GeneralBox,
+    pub bitcoind: BitcoindBox,
+    pub stakeholder_xpubs: StakeholderXpubsBox,
+    pub manager_xpubs: ManagerXpubsBox,
+    pub cosigner_keys: CosignerKeysBox,
+    pub manager: Option<ManagerBox>,
+    pub stakeholder: Option<StakeholderBox>,
+}
+
+#[derive(Debug, Clone)]
+pub struct GeneralBox {
+    /// The unvault output scripts relative timelock
+    pub unvault_csv: u32,
+    /// The host of the sync server (may be an IP or a hidden service)
+    pub coordinator_host: String,
+    /// The Noise static public key of the sync server
+    pub coordinator_noise_key: String,
+    /// The poll intervals for signature fetching (default: 1min)
+    pub coordinator_poll_seconds: Option<u64>,
+    /// An optional custom data directory
+    pub data_dir: Option<PathBuf>,
+    /// Whether to daemonize the process
+    pub daemon: Option<bool>,
+    /// What messages to log
+    pub log_level: Option<String>,
```

Comment:
I think it is a duplicate of what is in Config, 
maybe call Config in the body fn
`fn body<'a>(&self, cfg: &Config) -> Column<'a, Message>`